GHA Workflows using SYSELF_APP_PRIVATE_KEY fail in forks

/kind bug

**What steps did you take and what happened:**

I forked this repository to create pull requests (#551 #552). The workflow `Labels` failed in those PRs. Additionally, I get emails every 30 minutes from GitHub, that the workflow `Renovate` failed in my fork. Both fail at the same step:

```
2023-01-25T11:28:58.5203952Z ##[group]Run tibdex/github-app-token@v1
2023-01-25T11:28:58.5204230Z with:
2023-01-25T11:28:58.5204604Z   github_api_url: https://api.github.com
2023-01-25T11:28:58.5204991Z   repository: syself/cluster-api-provider-hetzner
2023-01-25T11:28:58.5205357Z ##[endgroup]
2023-01-25T11:28:58.6318613Z ##[error]Error: Input required and not supplied: app_id
```

**What did you expect to happen:**

I expected the CI pipeline for forks to run successfully, and to not receive notifications and emails every 30 minutes about a failed scheduled pipeline.

**Anything else you would like to add:**

The steps use secrets that are not available in my fork.

```
      - name: Generate Token
        uses: tibdex/github-app-token@v1
        id: generate-token
        with:
          app_id: ${{ secrets.SYSELF_APP_ID }}
          private_key: ${{ secrets.SYSELF_APP_PRIVATE_KEY }}
```


**Environment:**

Fork is at e957613d15cdd7754f988efc119a27b17f47dceb

--- a/.github/workflows/schedule-cache-cleaner-caph-image.yml
+++ b/.github/workflows/schedule-cache-cleaner-caph-image.yml
@@ -12,6 +12,7 @@ concurrency:
 jobs:
   cache-cleaner:
     runs-on: ubuntu-22.04
+    if: github.repository == 'syself/cluster-api-provider-hetzner'
     steps:
       # Load Golang cache build from GitHub
       - name: Load Caph Golang cache build from GitHub

--- a/.github/workflows/schedule-link-checker.yml
+++ b/.github/workflows/schedule-link-checker.yml
@@ -8,6 +8,7 @@ jobs:
   link-checker:
     name: Link Checker
     runs-on: ubuntu-latest
+    if: github.repository == 'syself/cluster-api-provider-hetzner'
     steps:
       - name: Checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1

--- a/.github/workflows/schedule-scan-image.yml
+++ b/.github/workflows/schedule-scan-image.yml
@@ -8,6 +8,7 @@ jobs:
   scan:
     name: Trivy
     runs-on: ubuntu-latest
+    if: github.repository == 'syself/cluster-api-provider-hetzner'
     container:
       image: ghcr.io/syself/caph-builder:1.0.10
       credentials:

